src/snowflake/snowpark/_internal/data_source/datasource_partitioner.py

```diff
@@ -0,0 +1,246 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+
+import datetime
+import decimal
+from _decimal import ROUND_HALF_EVEN, ROUND_HALF_UP
+from typing import Optional, Union, List, Callable
+
+import pytz
+from dateutil import parser
+from snowflake.snowpark._internal.data_source.utils import (
+    detect_dbms,
+    DBMS_MAP,
+    DRIVER_MAP,
+)
+
+from snowflake.snowpark._internal.data_source.datasource_reader import DataSourceReader
+import logging
```

Comment:
import logging should be placed before snowpark imports, not sure why our lint bot doesn't catch this